core/app/models/spree/store_credit.rb

```diff
@@ -184,7 +189,17 @@ def update_amount(amount, reason, user_performing_update)
     self.action = ADJUSTMENT_ACTION
     self.update_reason = reason
     self.action_originator = user_performing_update
-    save
+    if save
```

Comment:
that is what happens here, did not use `credit` method though, since there is already a `credit` method here.